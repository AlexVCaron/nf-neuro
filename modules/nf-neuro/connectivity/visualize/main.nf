process CONNECTIVITY_VISUALIZE {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:2.0.2' }"

    input:
    tuple val(meta), path(matrix), path (atlas_labels), path(labels_list)

    output:
    tuple val(meta), path("*.png"), emit: figure
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:

    String matrix_list = matrix.join(", ").replace(',', '')
    def args = ""
    if (atlas_labels) args += [" --lookup_table $atlas_labels "]
    if (labels_list) args += [" --labels_list $labels_list "]

    """
    for matrix in $matrix_list; do
        scil_viz_connectivity.py \$matrix \${matrix/.npy/_matrix.png} \
            --name_axis --display_legend --histogram \${matrix/.npy/_histogram.png} \
            --nb_bins 50 --exclude_zeros --axis_text_size 5 5 \
            $args
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    matrix_list = matrix.join(", ").replace(',', '')
    """
    for metric in $matrix_list; do
        base_name=\$(basename "\${metric}" .npy)
        touch "\${base_name}.png"
    done

    scil_viz_connectivity.py -h

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(pip list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
