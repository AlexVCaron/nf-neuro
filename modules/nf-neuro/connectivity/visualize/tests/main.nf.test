// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test connectivity/visualize
nextflow_process {

    name "Test Process CONNECTIVITY_VISUALIZE"
    script "../main.nf"
    process "CONNECTIVITY_VISUALIZE"

    tag "modules"
    tag "modules_nfcore"
    tag "connectivity"
    tag "connectivity/visualize"

    tag "subworkflows/load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( ["connectivity.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("connectivity - visualisation ") {
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/*.npy", checkIfExists: true),
                        [],
                        []
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
