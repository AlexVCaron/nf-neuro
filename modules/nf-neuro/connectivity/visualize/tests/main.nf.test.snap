{
    "connectivity - visualisation ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "afd_max_histogram.png:md5,acf59e8f729ae5b6c6d4862bff63498d",
                            "afd_max_matrix.png:md5,c2c06863a85fba3c675603d75e2f49a6",
                            "len_histogram.png:md5,d268e9f61ab4fd6b4ad16f327aef5c6a",
                            "len_matrix.png:md5,5d9665b12cd181ee7292cd5586da82bb",
                            "mask_histogram.png:md5,c343e261a0f19ceff2857e1df6064c22",
                            "mask_matrix.png:md5,c335c8f738b834dcdf2fa575ee7f5b0b",
                            "pval_histogram.png:md5,2046f815ad1bdd03916a00fa45e8cbbd",
                            "pval_matrix.png:md5,b48d02a53037e0919a9aa78b5c9d76fe",
                            "sc_add_10_histogram.png:md5,a00b70137e4672ac77fe88be3bdade34",
                            "sc_add_10_matrix.png:md5,eb7bbbadff0f0436e72af3036e13c282",
                            "sc_histogram.png:md5,1da6479ea5a4a62a38a1e306ab76dfe3",
                            "sc_lower_threshold_histogram.png:md5,2fbe3def2e032802ceac1181b633889d",
                            "sc_lower_threshold_matrix.png:md5,61ea6741af1cbac11d3b37fd1c72a617",
                            "sc_masked_histogram.png:md5,730faa50abaacc7521f5c7a1bd649fdf",
                            "sc_masked_matrix.png:md5,599a327daa5ddc18b68e5846f4562941",
                            "sc_matrix.png:md5,7925a60aa3579c93e0e7c3c7da9a219c",
                            "sc_norm_histogram.png:md5,3c049f3013317d2a547887eab6c6c713",
                            "sc_norm_matrix.png:md5,00a44b0ce567b5a874af4f72e3ccbf3a",
                            "sc_norm_vol_histogram.png:md5,25a78947709df35fd05ee078293100e2",
                            "sc_norm_vol_matrix.png:md5,64b1865183fd34a84e9a29099c3d6a6f",
                            "sc_normalize_histogram.png:md5,4746d4fc358c0602abd36567808d8651",
                            "sc_normalize_matrix.png:md5,a07a45fa844ff060d6f85bc6ad232a7d",
                            "sc_reo_fake_histogram.png:md5,2f098db8f5006438774c7c809ad21b78",
                            "sc_reo_fake_matrix.png:md5,abf3f196d7a2955551ec7259c5cadd60",
                            "sc_thr_histogram.png:md5,2fbe3def2e032802ceac1181b633889d",
                            "sc_thr_matrix.png:md5,61ea6741af1cbac11d3b37fd1c72a617",
                            "sim_histogram.png:md5,d16bcaa64d755db6186a7253a672957a",
                            "sim_matrix.png:md5,1e267e0b553700cb36bff6a600839ed8",
                            "vol_histogram.png:md5,29cf95685db9a1e537df3c77ea7f0bc7",
                            "vol_matrix.png:md5,5f357f6000d07dfb35185ec4cebfbf3f"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,405994012003d8a4506c810eaa403022"
                ],
                "figure": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "afd_max_histogram.png:md5,acf59e8f729ae5b6c6d4862bff63498d",
                            "afd_max_matrix.png:md5,c2c06863a85fba3c675603d75e2f49a6",
                            "len_histogram.png:md5,d268e9f61ab4fd6b4ad16f327aef5c6a",
                            "len_matrix.png:md5,5d9665b12cd181ee7292cd5586da82bb",
                            "mask_histogram.png:md5,c343e261a0f19ceff2857e1df6064c22",
                            "mask_matrix.png:md5,c335c8f738b834dcdf2fa575ee7f5b0b",
                            "pval_histogram.png:md5,2046f815ad1bdd03916a00fa45e8cbbd",
                            "pval_matrix.png:md5,b48d02a53037e0919a9aa78b5c9d76fe",
                            "sc_add_10_histogram.png:md5,a00b70137e4672ac77fe88be3bdade34",
                            "sc_add_10_matrix.png:md5,eb7bbbadff0f0436e72af3036e13c282",
                            "sc_histogram.png:md5,1da6479ea5a4a62a38a1e306ab76dfe3",
                            "sc_lower_threshold_histogram.png:md5,2fbe3def2e032802ceac1181b633889d",
                            "sc_lower_threshold_matrix.png:md5,61ea6741af1cbac11d3b37fd1c72a617",
                            "sc_masked_histogram.png:md5,730faa50abaacc7521f5c7a1bd649fdf",
                            "sc_masked_matrix.png:md5,599a327daa5ddc18b68e5846f4562941",
                            "sc_matrix.png:md5,7925a60aa3579c93e0e7c3c7da9a219c",
                            "sc_norm_histogram.png:md5,3c049f3013317d2a547887eab6c6c713",
                            "sc_norm_matrix.png:md5,00a44b0ce567b5a874af4f72e3ccbf3a",
                            "sc_norm_vol_histogram.png:md5,25a78947709df35fd05ee078293100e2",
                            "sc_norm_vol_matrix.png:md5,64b1865183fd34a84e9a29099c3d6a6f",
                            "sc_normalize_histogram.png:md5,4746d4fc358c0602abd36567808d8651",
                            "sc_normalize_matrix.png:md5,a07a45fa844ff060d6f85bc6ad232a7d",
                            "sc_reo_fake_histogram.png:md5,2f098db8f5006438774c7c809ad21b78",
                            "sc_reo_fake_matrix.png:md5,abf3f196d7a2955551ec7259c5cadd60",
                            "sc_thr_histogram.png:md5,2fbe3def2e032802ceac1181b633889d",
                            "sc_thr_matrix.png:md5,61ea6741af1cbac11d3b37fd1c72a617",
                            "sim_histogram.png:md5,d16bcaa64d755db6186a7253a672957a",
                            "sim_matrix.png:md5,1e267e0b553700cb36bff6a600839ed8",
                            "vol_histogram.png:md5,29cf95685db9a1e537df3c77ea7f0bc7",
                            "vol_matrix.png:md5,5f357f6000d07dfb35185ec4cebfbf3f"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,405994012003d8a4506c810eaa403022"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-17T18:47:12.144528447"
    },
    "connectivity - visualisation - stub-run": {
        "content": [
            [
                "versions.yml:md5,405994012003d8a4506c810eaa403022"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-17T18:47:28.843510255"
    }
}